src/auth/Crypto.cc

```diff
@@ -595,9 +595,9 @@ void CryptoKey::print(std::ostream &out) const
 void CryptoKey::to_str(std::string& s) const
 {
   int len = secret.length() * 4;
-  char buf[len];
-  hex2str(secret.c_str(), secret.length(), buf, len);
-  s = buf;
+  std::vector<char> buf(len);
```

Comment:
instead of always allocating, maybe consider [boost::container::small_vector](https://www.boost.org/doc/libs/1_69_0/doc/html/boost/container/small_vector.html) with a reasonable amount of static storage? or if T is char and ~20 is a reasonable size, just depend on std::string and its small string optimization

Reply:
Looking at the surrounding methods, I doubt `CryptoKey::to_str()` is used, so it could probably just be removed.